nemo/utils/exp_manager.py

```diff
@@ -928,29 +929,46 @@ def on_save_checkpoint(self, trainer, pl_module, checkpoint):
             # Load the best model and then re-save it
             app_state = AppState()
             if app_state.model_parallel_size is not None and app_state.model_parallel_size > 1:
-                raise ValueError(f'always_save_nemo is not implemented for model parallel models.')
+                logging.warning(f'always_save_nemo will slow down training for model_parallel > 1.')
             # since we are creating tarfile artifacts we need to update .nemo path
             app_state.model_restore_path = os.path.abspath(
                 os.path.expanduser(os.path.join(self.dirpath, self.prefix + self.postfix))
             )
             if self.save_best_model:
-                if not os.path.exists(self.best_model_path):
+                injected_best_model_path = inject_model_parallel_rank(self.best_model_path)
```

Comment:
Shouldn't this inject happen only if model parallel > 1?

Reply:
inject is a no-op when model parallel == 1. So i didn't want to add more if statements.